tests/dynamicconfig.go

```diff
@@ -75,10 +76,24 @@ func (d *dcClient) GetValue(name dynamicconfig.Key) []dynamicconfig.ConstrainedV
 	return d.fallback.GetValue(name)
 }
 
-func (d *dcClient) OverrideValue(name dynamicconfig.Key, value any) {
+// OverrideValue overrides a value for the duration of a test. Once the test completes
+// the previous value (if any) will be restored
+func (d *dcClient) OverrideValue(t *testing.T, name dynamicconfig.Key, value any) {
```

Comment:
maybe this should be like this for flexibility, so it can be used outside of testing:
```suggestion
type Cleanuper interface { Cleanup(func()) }
func (d *dcClient) OverrideValue(c Cleanuper, name dynamicconfig.Key, value any) {
```

Reply:
Do we have a reason to use this outside of testing?
If so, why not just take a `cleanup func()` and run it if not nil rather than requiring an interface for this

Reply:
sure, plain `func()` is fine too. just more typing at the call sites to add `.Cleanup`. but if there's only one call site since there's a wrapper in `FunctionalTestBase`, it's great

Reply:
I'm still concerned that we don't have a use for this outside of testing, and I'd like to leave it as is if its only for testing